libhfcommon/log.c

```diff
@@ -181,3 +181,18 @@ int logFd(void) {
 enum llevel_t logGetLevel(void) {
     return hf_log_level;
 }
+
+#if defined(__sun)
+void dprintf(int fd, const char *fmt, ...) {
+    va_list ap;
+    va_start(ap, fmt);
+    vdprintf(fd, fmt, ap);
+    va_end(ap);
+}
+
+int vdprintf(int fd, const char *fmt, va_list ap) {
+    FILE *fp;
+    fp = fdopen(fd, "w");
+    return vfprintf(fp, fmt, ap);
```

Comment:
Is fclose needed here?

Reply:
Sorry, I mean fclose would close the fd as well, but aren't we allocating struct for FILE* each time we do fdopen?

Reply:
May fdopen/fclose on dup'd fd?

Reply:
tried to avoid using the opaque filling approach here but ok may need fclose indeed

Reply:
tried but that faults.

Reply:
will try using the opaque field instead.